delete.go

```diff
@@ -0,0 +1,14 @@
+package influxdb
+
+import "context"
+
+// Predicate is something that can match on a series key.
+type Predicate interface {
```

Comment:
I'm wondering why this is needed. We have the same interface in `tsm1.Predicate`.